### PR TITLE
#156 validate and normalize windows includes

### PR DESCRIPTION
**Pull request by nittka**
fix for #156

### CODE REVIEW

File: org.elysium.parent/org.elysium.ui/src/org/elysium/ui/quickfix/LilyPondQuickfixProvider.java [hunk 1]
```diff
@@ -108,4 +109,24 @@ public void apply(EObject element, IModificationContext context) throws Exceptio
...
+	public void normalizeWindowsInclude(final Issue issue, IssueResolutionAcceptor acceptor) {
+		acceptor.accept(issue, "normalize include", null, null, new ISemanticModification() {
+
+			@Override
+			public void apply(EObject element, IModificationContext context) throws Exception {
+				if (element instanceof Include) {
+					String include = ((Include) element).getImportURI();
+					if(include!=null) {
+						if(include.startsWith("file:/")) {
+							include=include.substring(6);
```
Comment (thSoft) [inline org.elysium.parent/org.elysium.ui/src/org/elysium/ui/quickfix/LilyPondQuickfixProvider.java]: I propose using `replaceFirst` instead.

File: org.elysium.parent/org.elysium/src/org/elysium/validation/LilyPondValidator.java [hunk 2]
```diff
@@ -90,4 +90,16 @@ public void checkInclude(Include include) {
 			addIssue("Include with absolute location", getCurrentObject(), LilypondPackage.Literals.INCLUDE__IMPORT_URI, IssueCodes.ABSOLUTE_INCLUDE);
 		}
 	}
+
+	@Check
+	public void checkWindowsIncludeNormalized(Include include) {
+		if(LilyPondConstants.IS_WINDOWS && include.getImportURI()!=null) {
+			String includeString = include.getImportURI();
+			if(includeString.startsWith("file:/")) {
```
Comment (thSoft) [inline org.elysium.parent/org.elysium/src/org/elysium/validation/LilyPondValidator.java]: Please extract `"file:/"` to a constant and use it in `LilyPondQuickfixProvider` also.

File: org.elysium.parent/org.elysium/src/org/elysium/validation/LilyPondValidator.java [hunk 3]
```diff
@@ -90,4 +90,16 @@ public void checkInclude(Include include) {
...
+				error("file URIs are not allowed", LilypondPackage.Literals.INCLUDE__IMPORT_URI, 0, IssueCodes.WINDOWS_NORMALIZED_INCLUDE);
+			}else if(includeString.indexOf('\\')>=0){
```
Comment (thSoft) [inline org.elysium.parent/org.elysium/src/org/elysium/validation/LilyPondValidator.java]: I'd propose using `contains` to enhance readability.

**Comment by nittka** [comment]
cleanup done

**Comment by thSoft** [comment]
Thanks for the cleanup!

**Pull request merged**